metrics/ci_integrations/test/client/github_actions/test_utils/mock/github_actions_mock_server.dart

```diff
@@ -0,0 +1,296 @@
+
+    _setNextPageUrlHeader(request, workflowRuns.length);
+
+    workflowRuns = _paginate(workflowRuns, runsPerPage, pageNumber);
+
+    final _response = {
+      'total_count': workflowRuns.length,
+      'workflow_runs': workflowRuns.map((run) => run.toJson()).toList(),
+    };
+
+    await _writeResponse(request, _response);
+  }
+
+  /// Responses with a list of all workflow run jobs for a specific workflow run.
+  Future<void> _workflowRunJobsResponse(HttpRequest request) async {
+    final status = _extractRunStatus(request);
+    final runsPerPage = _extractPerPage(request);
+    final pageNumber = _extractPage(request);
+
+    List<WorkflowRunJob> workflowRunJobs = _generateWorkflowRunJobs(status);
+
+    _setNextPageUrlHeader(request, workflowRunJobs.length);
+
+    workflowRunJobs = _paginate(workflowRunJobs, runsPerPage, pageNumber);
+
+    final _response = {
+      'total_count': workflowRunJobs.length,
+      'jobs': workflowRunJobs.map((run) => run.toJson()).toList(),
+    };
+
+    await _writeResponse(request, _response);
+  }
+
+  /// Responses with a list of artifacts for a specific workflow run.
+  Future<void> _workflowRunArtifactsResponse(HttpRequest request) async {
+    final runsPerPage = _extractPerPage(request);
+    final pageNumber = _extractPage(request);
+    List<WorkflowRunArtifact> artifacts = _generateArtifacts();
+
+    _setNextPageUrlHeader(request, artifacts.length);
+
+    artifacts = _paginate(artifacts, runsPerPage, pageNumber);
+
+    final _response = {
+      'total_count': artifacts.length,
+      'artifacts': artifacts.map((artifact) => artifact.toJson()).toList(),
+    };
+
+    await _writeResponse(request, _response);
+  }
+
+  /// Redirects to the artifact download URL.
+  Future<void> _downloadArtifactResponse(HttpRequest request) async {
+    final uri = Uri.parse(url);
+
+    await request.response.redirect(
+      Uri(host: uri.host, port: uri.port, path: _downloadPath),
+      status: HttpStatus.found,
+    );
+
+    await request.response.close();
+  }
+
+  /// Returns a [Uint8List] to emulate download.
+  Future<void> _downloadResponse(HttpRequest request) async {
+    await _writeResponse(request, Uint8List.fromList([]));
+  }
+
+  /// Adds a [HttpStatus.notFound] status code to the [HttpRequest.response]
+  /// and closes it.
+  Future<void> _notFoundResponse(HttpRequest request) async {
+    request.response.statusCode = HttpStatus.notFound;
+
+    await request.response.close();
+  }
+
+  /// Chunks the given [items], limiting to the [limit],
+  /// starting from the [pageIndex].
+  List<T> _paginate<T>(List<T> items, [int limit, int pageIndex]) {
+    if (limit != null && pageIndex != null) {
+      final from = (pageIndex - 1) * limit;
+
+      return items.skip(from).take(limit).toList();
+    } else if (limit != null) {
+      return items.take(limit).toList();
+    }
+
+    return items;
+  }
+
+  /// Generates a list of [WorkflowRun], filtered by the given [status].
```

Comment:
```suggestion
  /// Generates a list of [WorkflowRun] with the given [status].
  ///
  /// If the given [status] is null, the [GithubActionStatus.completed] is used.
```
The same for the `_generateWorkflowRunJobs ` method.